src/tools/perf/ucp_tests.cc

```diff
@@ -286,16 +288,28 @@ class ucp_perf_test_runner {
         remote_addr = m_perf.ucp.peers[1 - my_index].remote_addr + m_perf.offset;
         rkey        = m_perf.ucp.peers[1 - my_index].rkey;
         sn          = 0;
+        if (UCX_PERF_CMD_TAG == CMD) {
```

Comment:
code duplication - i'd suggest to add m_send_datatype,m_recv_datatype variables and initialize them in a common function. Then, set local variables send_datatype and recv_datatype to the member variables in this function